trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/rest/TrinoLogsResource.java

```diff
@@ -0,0 +1,205 @@
+            requireNonNull(logStreamName, "logStreamName is null");
+
+            if (logStreamNames.isEmpty() && logStreamName.isEmpty()) {
+                logStreamNames = Optional.of(DEFAULT_STREAMS);
+            }
+        }
+    }
+
+    // see https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_GetLogEvents.html
+    public record GetLogEventsResponse(List<Event> events, Optional<String> nextBackwardToken, Optional<String> nextForwardToken)
+    {
+        public GetLogEventsResponse
+        {
+            requireNonNull(events, "events is null");
+            requireNonNull(nextBackwardToken, "nextBackwardToken is null");
+            requireNonNull(nextForwardToken, "nextForwardToken is null");
+        }
+
+        public record Event(String logStreamName, String eventId, long ingestionTime, long timestamp, String message)
+        {
+            public Event
+            {
+                requireNonNull(logStreamName, "logStreamName is null");
+                requireNonNull(eventId, "eventId is null");
+                requireNonNull(message, "message is null");
+            }
+        }
+    }
+
+    @Inject
+    public TrinoLogsResource(RequestLoggerController loggerController, ObjectMapper objectMapper)
+    {
+        this.loggerController = requireNonNull(loggerController, "loggerController is null");
+        this.objectMapper = requireNonNull(objectMapper, "objectMapper is null");
+    }
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response processRequest(@Context Request request, @HeaderParam("X-Amz-Target") String target)
+    {
+        String command = Splitter.on('.').trimResults().limit(2).splitToList(target).getLast().toLowerCase(Locale.ROOT);
+        return switch (command) {
+            // filter/get log events are basically the same.
+            // We've made GetLogEventsRequest a union of both and one method can satisfy both
+            case "filterlogevents", "getlogevents" -> getLogEvents(request);
+            default -> Response.status(BAD_REQUEST).build();
+        };
+    }
+
+    private Response getLogEvents(Request request)
+    {
+        byte[] bytes = request.requestContent().standardBytes().orElseThrow(() -> new WebApplicationException(BAD_REQUEST));
+        GetLogEventsRequest getLogEvents;
+        try {
+            getLogEvents = objectMapper.readValue(bytes, GetLogEventsRequest.class);
+        }
+        catch (IOException e) {
+            throw new WebApplicationException(e, BAD_REQUEST);
+        }
+
+        // TODO - support filter patterns
+
+        boolean startFromHead = getLogEvents.startFromHead.orElse(false);
+        Predicate<SaveEntry> predicate = entry -> startTimePasses(getLogEvents, entry) &&
+                endTimePasses(getLogEvents, entry) &&
+                logStreamNamesPasses(entry, getLogEvents);
+        int limit = getLogEvents.limit.orElse(Integer.MAX_VALUE);
```

Comment:
My goal was not to emulate this API exactly. We can limit it per the doc but it seemed more usable to not. wdyt?

Reply:
I'm not that concerned with limiting to be at most 10,000 log events, but perhaps `Integer.MAX_VALUE` as the default is too large?

Anyone who sends a request without specifying a limit may be getting a very very large response

Reply:
There's an internal config'd limit already. See: https://github.com/trinodb/aws-proxy/pull/120/files#diff-d554196acd7df8eae8ba40d4facd4de1b2b917793b097f5b965b2280e3d1df78R36

Reply:
I saw that one as part of #119 , but the config can be set to a much higher number. And particularly in the future I imagine it would be possible to store a higher number of entries in an external store (not in memory), so IMHO it would make sense to apply a limit to the number of entries returned per request.

No strong opinion on this one, just my preference but if you strongly want to avoid adding that limit here I'm happy to ignore this for now

Reply:
I'd prefer to leave it as is.